src/core/thread/mle.cpp

```diff
@@ -1583,7 +1595,14 @@ void Mle::HandleStateChanged(otChangedFlags aFlags)
 
     if ((aFlags & (OT_CHANGED_IP6_MULTICAST_SUBSCRIBED | OT_CHANGED_IP6_MULTICAST_UNSUBSCRIBED)) != 0)
     {
+#if (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2)
```

Comment:
What about we add a new method for judging if the node needs to register MAs so that we can avoid using different conditions for different thread versions, making the code more compact?

Reply:
added a private method.